Makefile

```diff
@@ -0,0 +1,95 @@
+CWD ?= CURRENT_WORKING_DIRECTIONRY_NOT_SUPPLIED
```

Comment:
I added a TODO at the top of the `Makefile`. 

I don't want this discussion (which is not a small discussion) to block this PR since it's a bit out of scope of what this tried to do, but we could discuss this in another Github issue or protocol hour instead.

Reply:
I am hard pressed to see code added in this PR as out of scope for this PR.